src/main/java/org/folio/rest/converter/VendorConverter.java

```diff
@@ -45,4 +41,25 @@ private Providers convertVendor(Vendor vendor) {
         .withProviderToken(token != null ? new Token().withValue(token) : null))
       .withRelationships(EMPTY_PACKAGE_RELATIONSHIP);
   }
+
+  public Provider convertToVendor(VendorById vendor) {
+    String token = vendor.getVendorToken();
+    return new Provider()
+      .withData(new ProviderData()
+        .withId(String.valueOf(vendor.getVendorId()))
+        .withType(PROVIDERS_TYPE)
+        .withAttributes(new ProviderDataAttributes()
+          .withName(vendor.getVendorName())
+          .withPackagesTotal(vendor.getPackagesTotal())
+          .withPackagesSelected(vendor.getPackagesSelected())
+          .withSupportsCustomPackages(vendor.isCustomer())
+          .withProviderToken(token != null ? new Token().withValue(token) : null)
```

Comment:
Here for "providerToken" object - it has the following attributes - "factName", "prompt", "helpText" and "value" and we want to populate all - not just value - this is a bit different than "providers" endpoint. Unfortunately, RM API does not give consistent results for collection and detailed record in all cases.

Reply:
To observe the difference mentioned above - you can play around with GET /custid/vendors and GET /custid/vendors/vendorId in this interactive docs - https://developer.ebsco.com/interactive/holdingsiq

Reply:
thank you for review Sobha. Fixed it by changing  "vendorToken" property from String to Object including "factName", "prompt", "helpText" and "value" parameters.
<img width="653" alt="screen shot 2018-11-05 at 2 34 25 pm" src="https://user-images.githubusercontent.com/37537790/47998478-20bc4f00-e108-11e8-80e1-c54829491127.png">
<img width="633" alt="screen shot 2018-11-05 at 2 35 32 pm" src="https://user-images.githubusercontent.com/37537790/47998486-26199980-e108-11e8-9166-6ff640181b0c.png">

Reply:
👍